packages/@aws-cdk/aws-cloudwatch/lib/metric.ts

```diff
@@ -251,8 +251,13 @@ export class Metric implements IMetric {
     });
   }
 
-  /** Dimensions of this metric */
+  /** 
+   * Dimensions of this metric 
+   * @deprecated Use 'dimensionsMap' instead
+   */
   public readonly dimensions?: DimensionHash;
+  /** Dimensions of this metric */
+  public readonly dimensionsMap?: DimensionsMap;
```

Comment:
You're creating this new prop, but not doing anything with it yet so it will always be undefined. There are a few other places where we use `this.dimensions` that you'll need to account for with this new prop if we are to deprecate the old prop